res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,736 @@
+            // Set eject back to 0 to turn off the eject button on screen
+            engine.setValue(group, "eject", 0);
+            loadButtonLongPressed[group] = false;
+        }
+    }
+};
+
+// Sampler functions
+// ========================================================
+ReloopBeatmix24.SamplerPad = function(channel, control, value, status, group) {
+    if (value === DOWN) {
+        if (engine.getValue(group, "track_samples")) { //Sampler loaded (playing or not)
+            engine.setValue(group, "cue_gotoandplay", 1);
+        } else {
+            engine.setValue(group, "LoadSelectedTrack", 1);
+        }
+    }
+};
+
+ReloopBeatmix24.ShiftSamplerPad = function(channel, control, value, status,
+    group) {
+    if (value === DOWN) {
+        if (engine.getValue(group, "track_samples")) { //Sampler loaded (playing or not)
+            if (engine.getValue(group, "play")) { // Sampler is playing
+                engine.setValue(group, "cue_gotoandstop", 1);
+            } else {
+                engine.setValue(group, "eject", 1);
+            }
+        } else {
+            engine.setValue(group, "LoadSelectedTrack", 1);
+        }
+    } else { // UP
+        if (!engine.getValue(group, "track_samples")) { // if empty
+            // Set eject back to 0 to turn off the eject button on screen
+            engine.setValue(group, "eject", 0);
+        }
+    }
+};
+
+ReloopBeatmix24.SamplerVol = function(channel, control, value, status, group) {
+    for (var i = 1; i <= engine.getValue("[Master]", "num_samplers"); i++) {
+        engine.setValue("[Sampler" + i + "]", "volume", value / 127.0);
+    }
+};
+
+// Jog Wheel functions
+// ========================================================
+
+ReloopBeatmix24.WheelTouch = function(channel, control, value, status, group) {
+    var deck = parseInt(group.substr(8, 1), 10);
+    if (value === DOWN) {
+        var alpha = 1.0 / 8;
+        var beta = alpha / 32;
+        engine.scratchEnable(deck, 800, JogRPM, alpha, beta);
+    } else {
+        engine.scratchDisable(deck);
+    }
+};
+
+ReloopBeatmix24.WheelTurn = function(channel, control, value, status, group) {
+    var newValue = value - 64;
+    var deck = parseInt(group.substr(8, 1), 10);
+
+    // In either case, register the movement
+    if (engine.isScratching(deck)) {
+        engine.scratchTick(deck, newValue); // Scratch!
+    } else {
+        engine.setValue(group, 'jog', newValue); // Pitch bend
+    }
+};
+
+// Led Feedback functions
+// ========================================================
+ReloopBeatmix24.AllJogLEDsToggle = function(deck, state, step) {
+    var step = typeof step !== 'undefined' ? step : 1; // default value
+    for (var j = 0x30; j <= 0x3F; j += step) {
+        midi.sendShortMsg(deck, j, state);
+    }
+};
+
+ReloopBeatmix24.deckLoaded = function(value, group, control) {
+    var i;
+    switch (group.substr(1, 7)) {
+        case "Channel":
+            var channelRegEx = /\[Channel(\d+)\]/;
```

Comment:
I'm not sure if this makes a difference, but I think it may be better for performance to define this regex as a global or property of ReloopBeatmix24 outside of any functions so the JS engine doesn't have to compile the regex on each function run.